File: Win/ui/views/team_leader_dashboard_view.py
"""
TeamLeaderDashboardView.py — DAS CRM Windows
Team Leader-Specific KPIs and Direct Report Performance
Feature parity with Android TeamLeaderDashboardScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QScrollArea,
    QFrame, QGridLayout, QTableWidget, QTableWidgetItem, QAbstractItemView
)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont, QBrush, QColor
from dataclasses import dataclass

# ─────────────────────────────────────────────────────────────────────────────────────
# DATA MODELS
# ─────────────────────────────────────────────────────────────────────────────────────

@dataclass
class DirectReport:
    """Team leader's direct report"""
    id: str
    name: str
    leadsAssigned: int
    leadsConverted: int
    conversionRate: str  # percentage
    thisMonthRevenue: str
    quota: str
    quotaAttainment: int  # 0-100%
    status: str  # ON_TRACK, AT_RISK, CRITICAL

FALLBACK_DIRECT_REPORTS = [
    DirectReport("dr1", "Rajesh Kumar", 18, 5, "27.8%", "₹2,40,000", "₹3,50,000", 68, "ON_TRACK"),
    DirectReport("dr2", "Priya Sharma", 12, 4, "33.3%", "₹1,80,000", "₹3,00,000", 60, "AT_RISK"),
    DirectReport("dr3", "Amit Patel", 15, 3, "20.0%", "₹1,50,000", "₹3,20,000", 47, "CRITICAL"),
]

# ─────────────────────────────────────────────────────────────────────────────────────
# STAT CARD COMPONENT
# ─────────────────────────────────────────────────────────────────────────────────────

class StatCard(QFrame):
    """KPI metric display card"""
    def __init__(self, label: str, value: str, icon: str, color: str, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 16px;
            }
        """)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(8)

        # Header
        headerLayout = QHBoxLayout()
        iconLabel = QLabel(icon)
        iconLabel.setFont(QFont("Segoe UI", 18))
        headerLayout.addWidget(iconLabel)

        labelWidget = QLabel(label)
        labelWidget.setFont(QFont("Segoe UI", 10))
        labelWidget.setStyleSheet("color: #94a3b8;")
        headerLayout.addWidget(labelWidget)
        headerLayout.addStretch()

        layout.addLayout(headerLayout)

        # Value
        valueLabel = QLabel(value)
        valueLabel.setFont(QFont("Segoe UI", 18, QFont.Weight.Bold))
        valueLabel.setStyleSheet(f"color: {color};")
        layout.addWidget(valueLabel)

        layout.addStretch()

# ─────────────────────────────────────────────────────────────────────────────────────
# MAIN TEAM LEADER DASHBOARD VIEW
# ─────────────────────────────────────────────────────────────────────────────────────

class TeamLeaderDashboardView(QWidget):
    """Team Leader Dashboard with Direct Report Metrics"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
        """)

        self.directReports = list(FALLBACK_DIRECT_REPORTS)
        self._build_ui()

    def _build_ui(self):
        """Build team leader dashboard UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(12)

        # Title
        titleLabel = QLabel("🏆 Team Leader Dashboard")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(titleLabel)

        # KPI Grid
        kpiLayout = QGridLayout()
        kpiLayout.setContentsMargins(0, 0, 0, 0)
        kpiLayout.setSpacing(12)

        totalLeads = sum(dr.leadsAssigned for dr in self.directReports)
        totalConverted = sum(dr.leadsConverted for dr in self.directReports)
        avgConvRate = f"{(totalConverted/totalLeads*100):.1f}%" if totalLeads > 0 else "0%"
        totalRevenue = sum(int(dr.thisMonthRevenue.replace("₹", "").replace(",", "")) for dr in self.directReports)

        kpiCards = [
            ("Direct Reports", str(len(self.directReports)), "👥", "#818cf8"),
            ("Total Leads", str(totalLeads), "📊", "#60a5fa"),
            ("Avg Conversion", avgConvRate, "📈", "#34d399"),
            ("Team Revenue", f"₹{totalRevenue:,}", "💰", "#fbbf24"),
        ]

        for idx, (label, value, icon, color) in enumerate(kpiCards):
            card = StatCard(label, value, icon, color)
            kpiLayout.addWidget(card, idx // 2, idx % 2, 1, 1)

        scrollLayout.addLayout(kpiLayout)

        # Direct Reports Section
        repTitleLabel = QLabel("👥 Direct Reports Performance")
        repTitleLabel.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        repTitleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(repTitleLabel)

        # Direct Reports Table
        self.reportsTable = QTableWidget()
        self.reportsTable.setColumnCount(8)
        self.reportsTable.setHorizontalHeaderLabels([
            "Name", "Leads", "Converted", "Conv. %", "Revenue", "Quota", "Attainment %", "Status"
        ])
        self.reportsTable.horizontalHeader().setStretchLastSection(False)
        self.reportsTable.setSelectionBehavior(QAbstractItemView.SelectionBehavior.SelectRows)
        self.reportsTable.setSelectionMode(QAbstractItemView.SelectionMode.SingleSelection)
        self.reportsTable.setStyleSheet("""
            QTableWidget { background-color: #030712; gridline-color: #1e293b; }
            QHeaderView::section { background-color: #0b1329; color: #818cf8; padding: 6px;
                                   border: none; border-right: 1px solid #1e293b; }
        """)

        self.reportsTable.setColumnWidth(0, 140)
        self.reportsTable.setColumnWidth(1, 70)
        self.reportsTable.setColumnWidth(2, 90)
        self.reportsTable.setColumnWidth(3, 80)
        self.reportsTable.setColumnWidth(4, 110)
        self.reportsTable.setColumnWidth(5, 110)
        self.reportsTable.setColumnWidth(6, 120)
        self.reportsTable.setColumnWidth(7, 100)

        self._refresh_reports_table()

        scrollLayout.addWidget(self.reportsTable, 1)

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _refresh_reports_table(self):
        """Refresh direct reports table"""
        self.reportsTable.setRowCount(len(self.directReports))

        for rowIdx, report in enumerate(self.directReports):
            self.reportsTable.setItem(rowIdx, 0, QTableWidgetItem(report.name))
            self.reportsTable.setItem(rowIdx, 1, QTableWidgetItem(str(report.leadsAssigned)))
            self.reportsTable.setItem(rowIdx, 2, QTableWidgetItem(str(report.leadsConverted)))
            self.reportsTable.setItem(rowIdx, 3, QTableWidgetItem(report.conversionRate))
            self.reportsTable.setItem(rowIdx, 4, QTableWidgetItem(report.thisMonthRevenue))
            self.reportsTable.setItem(rowIdx, 5, QTableWidgetItem(report.quota))

            # Attainment %
            attainmentItem = QTableWidgetItem(f"{report.quotaAttainment}%")
            attainmentColor = "#34d399" if report.quotaAttainment >= 80 else "#fbbf24" if report.quotaAttainment >= 60 else "#ef4444"
            attainmentItem.setForeground(QBrush(QColor(attainmentColor)))
            self.reportsTable.setItem(rowIdx, 6, attainmentItem)

            # Status
            statusItem = QTableWidgetItem(report.status)
            statusColor = "#34d399" if report.status == "ON_TRACK" else "#fbbf24" if report.status == "AT_RISK" else "#ef4444"
            statusItem.setForeground(QBrush(QColor(statusColor)))
            self.reportsTable.setItem(rowIdx, 7, statusItem)

File: macOs/Sources/DASCRM/Views/TeamLeader/TeamLeaderDashboardView.swift
"""
TeamLeaderDashboardView.swift — DAS CRM macOS
Team Leader-Specific KPIs and Direct Report Performance
Feature parity with Android TeamLeaderDashboardScreen.tsx
"""

import SwiftUI

struct DirectReport: Identifiable {
    let id: String
    let name: String
    let leadsAssigned: Int
    let leadsConverted: Int
    let conversionRate: String
    let thisMonthRevenue: String
    let quota: String
    let quotaAttainment: Int
    let status: String
}

let fallbackDirectReports = [
    DirectReport(id: "dr1", name: "Rajesh Kumar", leadsAssigned: 18, leadsConverted: 5,
                conversionRate: "27.8%", thisMonthRevenue: "₹2,40,000", quota: "₹3,50,000",
                quotaAttainment: 68, status: "ON_TRACK"),
    DirectReport(id: "dr2", name: "Priya Sharma", leadsAssigned: 12, leadsConverted: 4,
                conversionRate: "33.3%", thisMonthRevenue: "₹1,80,000", quota: "₹3,00,000",
                quotaAttainment: 60, status: "AT_RISK"),
    DirectReport(id: "dr3", name: "Amit Patel", leadsAssigned: 15, leadsConverted: 3,
                conversionRate: "20.0%", thisMonthRevenue: "₹1,50,000", quota: "₹3,20,000",
                quotaAttainment: 47, status: "CRITICAL"),
]

@MainActor
class TeamLeaderDashboardViewModel: ObservableObject {
    @Published var directReports: [DirectReport] = fallbackDirectReports

    var totalDirectReports: Int {
        directReports.count
    }

    var totalLeads: Int {
        directReports.reduce(0) { $0 + $1.leadsAssigned }
    }

    var totalConverted: Int {
        directReports.reduce(0) { $0 + $1.leadsConverted }
    }

    var avgConversionRate: String {
        let total = totalLeads
        let converted = totalConverted
        guard total > 0 else { return "0%" }
        let rate = Double(converted) / Double(total) * 100
        return String(format: "%.1f%%", rate)
    }

    var teamRevenue: String {
        let total = directReports.reduce(0) { acc, report in
            let value = Int(report.thisMonthRevenue.replacingOccurrences(of: "₹", with: "").replacingOccurrences(of: ",", with: "")) ?? 0
            return acc + value
        }
        return "₹\(String(format: "%,d", total))"
    }
}

struct TeamLeaderDashboardView: View {
    @StateObject private var viewModel = TeamLeaderDashboardViewModel()

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            VStack(alignment: .leading, spacing: 12) {
                Text("🏆 Team Leader Dashboard")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.white)
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)

            // KPI Grid (2x2)
            VStack(alignment: .leading, spacing: 12) {
                HStack(spacing: 12) {
                    MetricCard(
                        icon: "👥",
                        label: "Direct Reports",
                        value: String(viewModel.totalDirectReports),
                        color: Color(red: 0.50, green: 0.45, blue: 0.98)
                    )

                    MetricCard(
                        icon: "📊",
                        label: "Total Leads",
                        value: String(viewModel.totalLeads),
                        color: Color(red: 0.38, green: 0.65, blue: 0.98)
                    )
                }

                HStack(spacing: 12) {
                    MetricCard(
                        icon: "📈",
                        label: "Avg Conversion",
                        value: viewModel.avgConversionRate,
                        color: Color(red: 0.2, green: 0.83, blue: 0.60)
                    )

                    MetricCard(
                        icon: "💰",
                        label: "Team Revenue",
                        value: viewModel.teamRevenue,
                        color: Color(red: 0.98, green: 0.75, blue: 0.14)
                    )
                }
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))

            // Direct Reports Section
            VStack(alignment: .leading, spacing: 12) {
                Text("👥 Direct Reports Performance")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundColor(.white)

                // Reports Table
                Table(viewModel.directReports) {
                    TableColumn("Name", value: \.name)
                    TableColumn("Leads") { report in
                        Text(String(report.leadsAssigned))
                    }
                    TableColumn("Converted") { report in
                        Text(String(report.leadsConverted))
                    }
                    TableColumn("Conv. %", value: \.conversionRate)
                    TableColumn("Revenue", value: \.thisMonthRevenue)
                    TableColumn("Quota", value: \.quota)
                    TableColumn("Attainment %") { report in
                        Text("\(report.quotaAttainment)%")
                            .foregroundColor(
                                report.quotaAttainment >= 80 ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                                report.quotaAttainment >= 60 ? Color(red: 0.98, green: 0.75, blue: 0.14) :
                                Color(red: 0.98, green: 0.30, blue: 0.40)
                            )
                    }
                    TableColumn("Status") { report in
                        Text(report.status)
                            .foregroundColor(
                                report.status == "ON_TRACK" ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                                report.status == "AT_RISK" ? Color(red: 0.98, green: 0.75, blue: 0.14) :
                                Color(red: 0.98, green: 0.30, blue: 0.40)
                            )
                    }
                }
                .background(Color(red: 0.03, green: 0.04, blue: 0.07))
                .onDoubleClickSelectAll(false)
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Team Leader")
    }
}

struct MetricCard: View {
    let icon: String
    let label: String
    let value: String
    let color: Color

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(icon)
                    .font(.system(size: 18))

                VStack(alignment: .leading, spacing: 2) {
                    Text(label)
                        .font(.system(size: 10))
                        .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                }

                Spacer()
            }

            Text(value)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(color)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(Color(red: 0.07, green: 0.09, blue: 0.16))
        .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
        .cornerRadius(12)
    }
}

#Preview {
    NavigationView {
        TeamLeaderDashboardView()
    }
}
